maps/map06.txt

```diff
@@ -1,3 +1,11 @@
-############
-#o  .     g#
-############
+###########
+#.. ... ..#
+### ### # #
+    ...
+#.# ### #.#
+#.# .o. #.#
+#.# ### #.#
+    ...
+# # ### ###
+#.. ... .g#
+###########
```

Comment:
Initial version was too hard for me, so I had to introduce some additional gapes to top and bottom sections to easily escape from them.